src/main/scala/SerialAdapter.scala

```diff
@@ -257,7 +286,30 @@ class SimSerial(w: Int) extends BlackBox with HasBlackBoxResource {
   addResource("/testchipip/csrc/testchip_tsi.h")
 }
 
-case class SerialTLParams(memParams: MasterPortParams, isMemoryDevice: Boolean = false, width: Int = 4, asyncResetQueue: Boolean = false)
+case class AXIClockParams(
+  clockFreqMHz: Double = 1000.0, // Match FireSim's 1GHz MBUS freq.
+  crossingType: ClockCrossingType = AsynchronousCrossing() // Default to async crossing
+)
+case class AXIMemOverSerialTLClockParams(
+  axiClockParams: Option[AXIClockParams] = Some(AXIClockParams()) // if set, axi port in different clk domain
+) {
+  def getMemFrequency(system: HasTileLinkLocations)(implicit p: Parameters): Double = {
+    axiClockParams match {
+      case Some(clkParams) => clkParams.clockFreqMHz * (1000 * 1000)
+      case None => {
+        // get freq. from what the master of the serial link specifies
```

Comment:
Isn't this getting the frequency from what the serial link masters rather than what masters the serial link?

Reply:
Oh yea this comment is off. The serial link masters the FBUS and is slave to the MBUS. The code is still fine though.